components/automate-ui/src/app/entities/automate-settings/automate-settings.model.ts

```diff
@@ -1,73 +1,182 @@
+  ServiceGroupRemoveServices = 'serviceGroupRemoveServices',
+
+  // Client Runs
+  ClientRunsRemoveData = 'clientRunsRemoveData',
+  ClientRunsLabelMissing = 'clientRunsLabelMissing',
+  ClientRunsRemoveNodes = 'clientRunsRemoveNodes',
+
+  // Compliance
+  ComplianceRemoveReports = 'complianceRemoveReports',
+  ComplianceRemoveScans = 'complianceRemoveScans'
+}
+
+export enum JobCategories {
+  Infra = 'infra',
+  Compliance = 'compliance',
+  EventFeed = 'event_feed',
+  Services = 'services'
+}
+
+export enum InfraJobName {
+  MissingNodes = 'missing_nodes',
   MissingNodesForDeletion = 'missing_nodes_for_deletion',
+  DeleteNodes = 'delete_nodes',
+  PeriodicPurgeTimeseries = 'periodic_purge_timeseries'
+}
 
-  // DeleteNodes: Removes completely from elasticsearch nodes that
-  // have been marked for deletion
-  DeleteNodes = 'delete_nodes'
+// Actions and ConvergeHistory are nested, but contained inside
+// the InfraJobName of PeriodicPurgeTimeseries
+export enum NestedJobName {
+  ComplianceReports = 'compliance-reports',
+  ComplianceScans = 'compliance-scans',
+  Feed = 'feed',
+  Actions = 'actions',
+  ConvergeHistory = 'converge-history'
 }
 
 export class IngestJob {
-  running: boolean;
+  category: JobCategories;
   name: string;
+  nested_name?: string;
+  recurrence?: string;
   threshold: string;
-  every?: string;
-  lastRun?: Date;
-  nextRun?: Date;
-  lastElapsed?: Date;
-  startedOn?: Date;
+  disabled: boolean;
+  purge_policies?: any; // TODO needs specific type
+  older_than_days?: number;
+  last_elapsed?: Date;
+  next_due_at?: Date;
+  last_enqueued_at?: Date;
+  last_started_at?: Date;
+  last_ended_at?: Date;
 
-  constructor(respJob: RespJob) {
+  constructor(category: JobCategories, respJob: RespJob) {
     if (respJob !== null) {
-      this.running = respJob.running;
+      this.category = category;
       this.name = respJob.name;
-      this.every = respJob.every;
+      this.nested_name = respJob.nested_name;
+      this.disabled = respJob.disabled;
+      this.recurrence = respJob.recurrence;
       this.threshold = respJob.threshold;
-      this.lastRun = new Date(respJob.last_run);
-      this.nextRun = new Date(respJob.next_run);
-      this.lastElapsed = new Date(respJob.last_elapsed);
-      this.startedOn = new Date(respJob.started_on);
+      this.purge_policies = respJob.purge_policies;
+      this.last_elapsed = new Date(respJob.last_elapsed);
+      this.next_due_at = new Date(respJob.next_due_at);
+      this.last_enqueued_at = new Date(respJob.last_enqueued_at);
+      this.last_started_at = new Date(respJob.last_started_at);
+      this.last_ended_at = new Date(respJob.last_ended_at);
     }
   }
 }
+
+export class UnfurledJob {
+  disabled: boolean;
+  policy_name?: string;
+  older_than_days?: number;
+  name?: string;
+  threshold?: string;
+}
+
+// A JobRequestComponent is very flexible
+  // It needs to be able to contain an older API Job object
+  // And a newer API Job Object
```

Comment:
[ubernit] Never seen the comment tabbed out before.

Reply:
I generally tab out a comment when there is information that extends out from the first line, but by bringing this up, I'm realizing I could probably right this comment better so that that isn't required.  Good lookin out!